### ISSUE TITLE
Mixin crashing game despite being skipped

### ISSUE THREAD

**Issue by Krutoy242** [friendlyhj/ZenUtils#70]
I ran into test mod list with CT + ZenUtils and two other mods enabled.
I found that most mixins are obviously skipped:
```ruby
[CONSTRUCTING][CLIENT][TRACE] [mixin | SIDE_CLIENT]: Loading Script: {[0:mixin]: mixin\twilightforest.zs}
[CONSTRUCTING][CLIENT][INFO] Skip loading mixin class MixinBlockTFMagicLogSpecial, because the target twilightforest.block.BlockTFMagicLogSpecial is not found
[CONSTRUCTING][CLIENT][INFO] Skip loading mixin class MixinContainerTFUncrafting, because the target twilightforest.inventory.ContainerTFUncrafting is not found
```

[crafttweaker.log](https://github.com/user-attachments/files/17325617/crafttweaker.log)

But, just after those lines, game crashing on load:
```ruby
java.lang.ClassFormatError: Illegal class name "Ljava/lang/Object;" in class file Mixinmixin\Twilightforest_MixinContainerTFUncrafting_87
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(Unknown Source)
	at java.base/java.lang.ClassLoader.defineClass(Unknown Source)
	at stanhebben.zenscript.definitions.zenclasses.ParsedZenClass$1.find(ParsedZenClass.java:139)
	at stanhebben.zenscript.definitions.zenclasses.ParsedZenClass$1.access$000(ParsedZenClass.java:137)
	at stanhebben.zenscript.definitions.zenclasses.ParsedZenClass.writeClass(MixinParsedZenClass.java:141)
	at stanhebben.zenscript.ZenParsedFile.<init>(ZenParsedFile.java:142)
	at crafttweaker.runtime.CrTTweaker.loadScript(CrTTweaker.java:203)
	at crafttweaker.runtime.CrTTweaker.loadScript(CrTTweaker.java:96)
	at crafttweaker.runtime.CrTTweaker.loadScript(CrTTweaker.java:100)
	at crafttweaker.runtime.CrTTweaker.loadScript(CrTTweaker.java:78)
	at youyihj.zenutils.impl.runtime.ZenUtilsTweaker.loadScript(ZenUtilsTweaker.java:106)
```

[debug.log](https://github.com/user-attachments/files/17325628/debug.log)

I tried to add `#modloaded twilightforest` into file, but seems like this is not working and file not skipped for loading.

Expected behavior: game not crashing on erroring mixin script, or provide more precise error message about place where error is happens.

**Comment by Krutoy242** [comment]
This could be another issue, but `#modloaded` preprocessors not working in `#loader mixin`. This is not a disaster, since mostly affected test environment, but would be cool if it worked.

This what I have in the file:
```zs
#loader mixin
#modloaded actuallyadditions

import native.net.minecraft.block.state.IBlockState;
import native.net.minecraft.util.math.BlockPos;
import native.net.minecraft.init.Blocks;
import native.de.ellpeck.actuallyadditions.api.internal.IAtomicReconstructor;
import native.de.ellpeck.actuallyadditions.api.ActuallyAdditionsAPI;
import native.de.ellpeck.actuallyadditions.api.recipe.WeightedOre;

#mixin Mixin
```

Error:
```ruby
[CONSTRUCTING][CLIENT][TRACE] [mixin | SIDE_CLIENT]: Loading Script: {[0:mixin]: mixin\actuallyadditions.zs}
[CONSTRUCTING][CLIENT][ERROR] mixin\actuallyadditions.zs:7 > Not a valid type
```

And later in log i actually see how file is skipped:
```ruby
[PREINITIALIZATION][CLIENT][INFO] Ignoring script {[0:mixin]: mixin\actuallyadditions.zs} due to the following #modloaded preprocessor settings: [{actuallyadditions}]
```

**Comment by friendlyhj** [comment]
I can not reproduce the crash, but I fix `#modloaded` doesn't work in mixin script loader, it should fix the crash.

**Comment by Krutoy242** [comment]
Yes, `#modloaded` is working now so crash isn't actual anymore.

**Comment by Krutoy242** [comment]
![image](https://github.com/user-attachments/assets/61fc8c57-21b2-4335-8d83-91fdd8b31e05)
Ok, seems like the reason was my temporary "fix" of type error for `zenutils 1.20.8`.
